File: MiniDOMBase/DMD/Anagrafica/Tasks/Handlers/ITaskHandler.cs

namespace minidom
{
    public partial class Anagrafica
    {
        /// <summary>
        /// Handler di una regola
        /// </summary>
        public interface ITaskHandler
        {

            /// <summary>
            /// Esegue la regola specificata sul task sorgente e restituisce il task generato
            /// </summary>
            /// <param name="taskSorgente"></param>
            /// <param name="regola"></param>
            /// <param name="parametri"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            TaskLavorazione Handle(TaskLavorazione taskSorgente, RegolaTaskLavorazione regola, string parametri);
        }
    }
}

File: MiniDOMBase/DMD/Anagrafica/Tasks/ITaskHandler.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
     
    Public Interface ITaskHandler

        ''' <summary>
        ''' Esegue la regola specificata sul task sorgente e restituisce il task generato
        ''' </summary>
        ''' <param name="taskSorgente"></param>
        ''' <param name="regola"></param>
        ''' <param name="parametri"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function Handle(ByVal taskSorgente As TaskLavorazione, ByVal regola As RegolaTaskLavorazione, ByVal parametri As String) As TaskLavorazione

    End Interface


End Class